components/autocomplete/index.js

```diff
@@ -162,15 +163,19 @@ export class Autocomplete extends Component {
 	}
 
 	select( option ) {
+		const { onReplace } = this.props;
 		const { open, range, query } = this.state;
-		const { onSelect } = open || {};
+		const { getOptionCompletion } = open || {};
 
 		this.reset();
 
-		if ( onSelect ) {
-			const replacement = onSelect( option.value, range, query );
-			if ( replacement !== undefined ) {
-				this.replace( range, replacement );
+		if ( getOptionCompletion ) {
+			const { action, value } = getOptionCompletion( option.value, range, query );
+
+			if ( 'replace-block' === action ) {
```

Comment:
`components` are UI primitives not specific to any screen. There should be no understanding of what a block is at this point.

Reply:
Good point. Changing this to 'replace' would skirt that. Is that enough? (It was @youknowriad's original suggestion anyway, but I momentarily forgot this should not be block-aware)